hibernate-core/src/test/java/org/hibernate/orm/test/jpa/criteria/mapjoin/MapJoinTestWithEmbeddable.java

```diff
@@ -65,12 +65,12 @@ public void testSelectingKeyOfMapJoin() {
 	public void testSelectingValueOfMapJoin() {
 		doInJPA( this::entityManagerFactory, entityManager -> {
 			CriteriaBuilder cb = entityManager.getCriteriaBuilder();
-			CriteriaQuery<Node> query = cb.createQuery( Node.class );
+			CriteriaQuery<BatchNodeMetadata> query = cb.createQuery( BatchNodeMetadata.class );
 			Root<Batch> root = query.from( Batch.class );
 
-			MapJoin nodes = (MapJoin) root.join( "batchNodeMetadata" );
+			MapJoin<Batch, Node, BatchNodeMetadata> nodes = (MapJoin) root.join( "batchNodeMetadata" );
 
-			query.select( nodes );
+			query.select( nodes.value() );
```

Comment:
This migrated testing case is wrong technically in v5 for the type of `nodes` doesn't match up with the specified query type (i.e. `Node`). Given that the other testing case has tested `nodes.key()`, this testing case was changed to select `nodes.value()`.
A good example to showcase the power of Java generics type checking. Java compiler will complain type inconsistency as long as we use generics type to `MapJoin`.